orchestrator/domain/helpers.py

```diff
@@ -0,0 +1,28 @@
+from typing import Iterable, Tuple, Union
+
+from more_itertools import first
+
+from orchestrator.types import filter_nonetype, get_origin_and_args, is_optional_type, is_union_type
+
+
+def _to_product_block_field_type_iterable(product_block_field_type: Union[type, Tuple[type]]) -> Iterable[type]:
+    """Return an iterable of types in the given product block field.
+
+    Notes:
+        There is a type-checking pattern (if optional, if union, if list/tuple, etc) which occurs multiple times in
+        base.py and other parts of the codebase. We should try to combine those into one function.
+        However, this function works based on field types as returned by
+        DomainModel._get_depends_on_product_block_types() so it is not directly applicable.
+    """
+    _origin, args = get_origin_and_args(product_block_field_type)
+
+    if is_optional_type(product_block_field_type):
```

Comment:
What is the reasoning for this? previous code only has the `is_union_type`.
A `Union[int, str, None]` is also a valid optional type and this would only return `[int]`. maybe add the `filter_nonetype` to the `is_union_type`? (since the only difference between `is_union_type` and `is_optional_type` is the `None`.

Reply:
I don't remember, but I think you're right. I'll try to remove it and in is_union_type filter out the nonetype

Reply:
Yeah it works fine. Also makes more sense this way. Good spot!